src/Graphics/OpenGLContext/GLSL/glsl_CombinerProgramBuilder.h

```diff
@@ -2,108 +2,122 @@
 #include <memory>
 #include <Combiner.h>
 #include <Graphics/OpenGLContext/opengl_GLInfo.h>
+#include "glsl_CombinerProgramUniformFactory.h"
+#include "glsl_ShaderPart.h"
 
 namespace graphics {
 	class CombinerProgram;
 }
 
-namespace opengl {
-	class CachedUseProgram;
-}
-
 namespace glsl {
 
-	class ShaderPart;
-	class CombinerInputs;
-	class CombinerProgramUniformFactory;
+	class TextureConvert {
+	public:
+		void setMode(u32 _mode) {
+			m_mode = _mode;
+		}
+
+		bool getBilerp1() const {
+			return (m_mode & 1) != 0;
+		}
+
+		bool getBilerp0() const {
+			return (m_mode & 2) != 0;
+		}
+
+		bool useYUVCoversion() const {
+			return (m_mode & 3) != 3;
+		}
+
+		bool useTextureFiltering() const {
+			return (m_mode & 3) != 0;
+		}
+
+	private:
+		u32 m_mode;
+	};
 
 	class CombinerProgramBuilder
 	{
 	public:
-		CombinerProgramBuilder(const opengl::GLInfo & _glinfo, opengl::CachedUseProgram * _useProgram);
-		~CombinerProgramBuilder();
+		CombinerProgramBuilder(const opengl::GLInfo & _glinfo, opengl::CachedUseProgram * _useProgram,
+			std::unique_ptr<CombinerProgramUniformFactory> _uniformFactory);
+		virtual ~CombinerProgramBuilder();
 
 		graphics::CombinerProgram * buildCombinerProgram(Combiner & _color, Combiner & _alpha, const CombinerKey & _key);
 
-		const ShaderPart * getVertexShaderHeader() const;
+		virtual const ShaderPart * getVertexShaderHeader() const = 0;
 
-		const ShaderPart * getFragmentShaderHeader() const;
+		virtual const ShaderPart * getFragmentShaderHeader() const = 0;
 
-		const ShaderPart * getFragmentShaderEnd() const;
+		virtual const ShaderPart * getFragmentShaderEnd() const = 0;
 
-		bool isObsolete() const;
+		virtual bool isObsolete() const = 0;
```

Comment:
Indent is unnecessary here. Please remove left indent for the code inside `namespace `brackets, as it is done in cpp file.